internal/workflow/processing.go

```diff
@@ -439,6 +440,16 @@ func (w *ProcessingWorkflow) SessionHandler(
 		return err
 	}
 
+	// Validate PREMIS.
+	activityOpts := withActivityOptsForRequest(sessCtx)
+	err := temporalsdk_workflow.ExecuteActivity(activityOpts, xmlvalidate.Name, xmlvalidate.Params{
+		XMLPath: filepath.Join(tinfo.TempPath, "data", "metadata", "premis.xml"),
+		XSDPath: "/home/enduro/premis.xsd",
+	}).Get(activityOpts, nil)
+	if err != nil {
+		return err
+	}
```

Comment:
We should check if the error is a missing `premis.xml` file (shouldn't be considered as an error in workflow) or other type of error (should fail the workflow).

Reply:
Not sure if it will require changes in https://github.com/artefactual-sdps/temporal-activities/blob/main/xmlvalidate/activity.go#L45. You may be able to check if it is an `os.IsNotExist(err)` error, but need to be careful about the file (it could be a missing XSD file).